pandas/_libs/src/klib/khash.h

```diff
@@ -282,32 +282,39 @@ static const double __ac_HASH_UPPER = 0.77;
 	extern khuint_t kh_put_##name(kh_##name##_t *h, khkey_t key, int *ret); \
 	extern void kh_del_##name(kh_##name##_t *h, khuint_t x);
 
+// From https://stackoverflow.com/a/1489985/621736
+#define PASTER(x,y) x ## y
+#define EVALUATOR(x, y) PASTER(x, y)
+#define KHASH_FUNCTION_POSTFIX_0 _set
+#define KHASH_FUNCTION_POSTFIX_1
+#define KHASH_FUNCTION_POSTFIX(kh_is_map) KHASH_FUNCTION_POSTFIX_##kh_is_map
+
```

Comment:
Why is this needed at all? khash supports sets and maps out of the box.

Reply:
It supports both but AFAICT the macros KHASH_SET_INIT_INT64 and KHASH_MAP_INIT_INT64 will ultimately both generate a set of functions named kh_init_int64, kh_delete_int64, etc...(see the kh_is_map argument to the function like macro).  Whatever macro you invoke most recently will dictate if the function you call uses map or set code, but that is handled entirely within the function body. As far as the API is concerned it would be ambiguous which implementation you are using without knowing which macro was invoked

Since we use both map and set implementations this is pretty tough and maybe impossible to control via Cython, so by making functions like kh_delete_int64_set it disambiguates it from the existing generated function that works with maps

Reply:
Probably the init-macro should be called with a different name-argument, like `KHASH_SET_INIT_INT64(int64set)` instead of `KHASH_SET_INIT_INT64(int64)` then the functions will be named `kh_init_int64set` and `kh_delete_int64set`.

Reply:
Ah...so much simpler. Let me give that a shot

Reply:
Thanks! Missed the "forest through the trees" on this - helped me get back on track